cmake/Utils.cmake

```diff
@@ -59,17 +59,24 @@ MACRO(ADD_TARGET_PROPERTY TARGET PROPERTY VALUE)
     ENDIF()
 ENDMACRO(ADD_TARGET_PROPERTY)
 
-MACRO(GET_APP_VERSION APP_DIR VERSION_MAJOR VERSION_MINOR VERSION_MAINTENANCE)
-    FILE(READ "${APP_DIR}/version" VERSION_RAW)
-    STRING(REGEX MATCHALL "[0-9]" VERSION_PARTS ${VERSION_RAW})
-    LIST(GET VERSION_PARTS 0 ${VERSION_MAJOR})
-    LIST(GET VERSION_PARTS 1 ${VERSION_MINOR})
-    LIST(GET VERSION_PARTS 2 ${VERSION_MAINTENANCE})
+MACRO(GET_APP_VERSION GIT_DESCRIBE VERSION_MAJOR VERSION_MINOR VERSION_MAINTENANCE)
+    STRING(REGEX MATCH "v([0-9]+)[.]([0-9]+)[.]([0-9]+)" GIT_DESCRIBE_MATCH ${GIT_DESCRIBE})
```

Comment:
I don't think cmake has predefined character classes like `\d`, at least it's not documented: https://cmake.org/cmake/help/v3.0/command/string.html

However, at first I tried using `\.` to match a literal `.`, and it seems that using backslashes in this regex cause visual studio to break with a "can't parse escape sequence" error... typical cmake nonsense!

I think the `[.]` bit will only match literal `.` characters though, which is what we want. Characters that normally have special meanings like `.` are just literals when used in a character class ,the only special one is `-` for ranges.